### DISCUSSION TITLE
Copilot memory leak + cpu spinning

### DISCUSSION THREAD

**Discussion by sigmaSd** [community/community#28571] [Copilot Conversations]
Recently copilot started to take a lot of memory, and spins the cpu

To reproduce I just open a simple javascript file, and just wait, the memory seems to grow infinitely

![image](https://user-images.githubusercontent.com/22427111/183248747-659e20ae-2e0e-471d-904d-2fc2c8deee92.png)

**meta**
copilot.vim 1.4.3

**Comment by sigmaSd** [comment]
copilot 1.4.2 seems to not have the issue, I guess the problem started with the latest commit.

**Comment by DevJMD** [comment]
Yup, can confirm there is a huge leak in the last release for CoPilot. Worked fine before. Now PHPStorm is stuttering like mad. Disabling it fixed the issue, but I assume rolling back to the previous version -- for now -- is a temporary fix.

This was on MacOS Monterey, using PHPStorm 2022.4.

Plugin: [GitHub Copilot](https://plugins.jetbrains.com/plugin/17718-github-copilot/versions)
Version: [1.1.29.1869](https://plugins.jetbrains.com/plugin/17718-github-copilot/versions/stable/206547)

Get the stable release (previous version for PHPStorm) here:
https://plugins.jetbrains.com/plugin/17718-github-copilot/versions/stable/190492

If the above release doesn't fix the issue, roll further back to version [1.1.23.1570](https://plugins.jetbrains.com/plugin/17718-github-copilot/versions/stable/177247) for now.

Hope this helps anyone using JetBrains.

**Comment by Loubaris** [comment]
Yeah, the last release sound's to be having some issues.

**Comment by sigmaSd** [comment]
Seems to fixed in nvim extension version 1.5.1

**Reply by jack-michaud** [reply]
> Seems to fixed in nvim extension version 1.5.1

Not so sure - version 1.5.1 and now 1.5.2 both seemed to have this issue on my machine. Running multiple instances of vim will end up crashing my computer unless I `pkill node`.

Every one in a while, copilot will start consuming 100% CPU as well. Disabling copilot for the time being...

**Comment by jack-michaud** [comment]
Even after I close my editor (`nvim`) it seems like the `node` process running the copilot agent continues to run. Are other people seeing this behavior?

**Reply by hamidzr** [reply]
related issue: https://github.com/community/community/discussions/30864

**Comment by wwzeng1** [comment]
I saw this as well - our company is developing "Cursor for JetBrains". If you're interested you can try it here: https://docs.sweep.dev/

**Comment by 99sono** [comment]
For me the issue is now gone.

I am using the latest version of the copilot plugin as well as the highest release of IntelliJ Gateway with the most recent backend update released by IntelliJ.  
I am quite happy right now.
